packages/design-system-react-native/src/components/Checkbox/Checkbox.tsx

```diff
@@ -71,14 +64,12 @@ export const Checkbox = forwardRef<{ toggle: () => void }, CheckboxProps>(
       ]).start();
     }, [scaleAnim]);
 
-    // Press handler: update state, notify parent, then bounce
+    // Press handler: notify parent, then bounce
     const handlePress = () => {
       if (isDisabled) {
         return;
       }
-      const next = !internalSelected;
-      setInternalSelected(next);
-      onChange?.(next);
+      onChange?.(!isSelected);
```

Comment:
For a checkbox, clicking it toggles the state. In an <input>, the browser automatically passes the new checked value. In React Native version of the Checkbox, we need to calculate the new value (!isSelected) and send it to the parent via onChange.